### PR TITLE
CLI Skeleton

### LINKED ISSUE #36: Add CLI skeleton
Create a basic CLI framework that can be used with subcommands.

### PR DESCRIPTION
**Pull request by dwhswenson**
Resolves #36.

- [x] Separate core OPS CLI and functionality
- [x] Separate OPS command plugin functionality
- [x] Implement OpenFE CLI with basic command and plugin functionality
- [x] Separate OPS reusable parameter functionality
- [x] Move OPS infrastructure to separate package
- [x] Add testing for OpenFE custom code

*  ~Implement OpenFE reusable parameters~ To keep this PR smaller, I'll come back to this in a future PR, which will just add a few initial examples.

### CODE REVIEW

File: openfecli/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+from .plugins import OFECommandPlugin
```
Comment (richardjgowers) [inline openfecli/__init__.py:1]: Probably worth putting our OpenFE MIT header thing here and all files

**Review by richardjgowers — COMMENTED** [review]
Looking good! A good simple example would just be printing out an AtomMapping

**Comment by dwhswenson** [comment]
Ready for review, with the following notes:

* `plugcli` installation will be moved from pip to conda-forge in the future (see `environment.yaml`)
* Currently missing 4 lines in coverage. One of these (the `pass` in `cli.py`) will be easy to add testing for once we have at least one permanently available subcommand. The others are in the temporary example file `commands/echo.py`; I see no need to test that, since we'll replace it soon with some actually functional code.

**Comment by dwhswenson** [comment]
> Looking good! A good simple example would just be printing out an AtomMapping

This is my plan for a follow-up PR. It will be easy for @mikemhenry to then extend that idea the nice visualization he's working on.

**Review by richardjgowers — APPROVED** [review]
LGTM, I like the splitting of pre to post library objects with `my_command` and `my_command_main`

**Pull request merged**